### DISCUSSION TITLE
Monitoramento de erros/bugs

### DISCUSSION THREAD

**Discussion by jdgabriel** [backend-br/forum#132] [Q&A] [answered]
Fala galera... blza?
Chega uma hora na vida de um `dev jr` que ele tem que aprender como fazer um monitoramento de erros e bugs que não seja pelo `console.log` (Nada contra, uso muito).

Poderia me indicar algum `recurso / blog / post` que possa me mostrar e explicar como funciona o monitoramento de erros em um projeto de grande porte?
 Estou lendo sobre o `prom-client` com `Grafana`, mas não estou conseguindo ver em como esses logs podem me ajudar? (tudo é gráfico kk)

**Comment by gustavofreze** [comment, marked as answer]
> Fala galera... blza? Chega uma hora na vida de um `dev jr` que ele tem que aprender como fazer um monitoramento de erros e bugs que não seja pelo `console.log` (Nada contra, uso muito).
> 
> Poderia me indicar algum `recurso / blog / post` que possa me mostrar e explicar como funciona o monitoramento de erros em um projeto de grande porte? Estou lendo sobre o `prom-client` com `Grafana`, mas não estou conseguindo ver em como esses logs podem me ajudar? (tudo é gráfico kk)

**Ferramentas**

Em questão de ferramentas, você tem muitas opções, isso só considerando as melhores ferramentas do mercado. Algumas delas, vão te disponibilizar vários recursos diferentes, e aí você tem que entender o que faz sentido pra você.

Além do Grafana que você citou, eu recomendaria dar uma estudada sobre:

- [Splunk](https://www.splunk.com/en_us/products.html)
- [Datadog](https://docs.datadoghq.com)
- [New Relic](https://docs.newrelic.com)

**E como essas ferramentas te ajudam?** 

Você vai ter logs, métricas, integrações com serviços de logs de plataformas de cloud e muitas outras coisas. Você vai conseguir ter uma visão de tudo da sua app e qualquer outra coisa que interaja com ela (outras apps, bancos de dados e etc).
A questão aqui é, você configurar os ambientes corretamente, e de acordo com a necessidade de negócio, criar os alertas, dashboards e etc.

Atualmente eu trabalho com New Relic e Splunk, e posso te dizer, que além da abordagem reativa (comum), conseguimos ter uma abordagem proativa, ou seja, há casos que conseguimos identificar uma intermitência em algum fluxo, e atuar, antes que ele venha a se tornar um erro/problema.
